### PR TITLE
AACT-439: As a user I should be able to edit a query that I created

### PR DESCRIPTION
**Pull request by Javier-Jimenez-18**
- Added a button/link on the show page to /queries/:id/edit

Screen Shot 2022-12-05 at 3 36 13 PM

Screen Shot 2022-12-05 at 3 55 44 PM

- Created an edit action in the Saved Queries controller that loads the query that we are editing. It checks to see if the currently logged in user created the query and if they did create it, then show the edit page.

Screen Shot 2022-12-05 at 3 40 26 PM

Screen Shot 2022-12-05 at 3 40 42 PM

- If the logged in user did not create the query, it redirects them to the index page for the saved queries and adds a flash message saying “You do not have permission to edit that query”. 

Screen Shot 2022-12-05 at 3 43 33 PM

Screen Shot 2022-12-05 at 3 44 18 PM

- Updated SavedQueriesController RSpec Controller test file.

Screen Shot 2022-12-05 at 3 17 53 PM

- The SavedQueriesController RSpec Request test file has some issues and will be investigated further.

### CODE REVIEW

**Review by kholtzman5610 — APPROVED** [review]
Good job with the test cases and screenshots. Looks good

**Review by micronix — CHANGES_REQUESTED** [review]
Looks good, just some small changes

**Comment by Javier-Jimenez-18** [comment]
- Modified SavedQueriesController index, edit, update, and destroy actions to check to see if the currently logged in user created the query. If not, it renders a not found error (404) page.
- Index shows all the queries that belong to the currently logged in user & all the queries that don't belong to the user and have Public set true. 

Screen Shot 2022-12-14 at 4 58 56 PM

Screen Shot 2022-12-14 at 4 59 57 PM

Screen Shot 2022-12-14 at 5 04 41 PM

Screen Shot 2022-12-14 at 5 04 51 PM

- Updated RSpec Controller test file. 

Screen Shot 2022-12-14 at 4 56 05 PM

- Updated RSpec Request test file. Each test case PASSES when run individually. Will have to investigate further to see why the test file has FAILURES when all test cases are run together.

- Cleaned up and refactored code.

**Comment by Javier-Jimenez-18** [comment]
All saved queries that are public, or that belong to the current user are shown in the index page. But, the current user can't go to the show page of another user's query, not_found (404) gets rendered. Is this what we want? I thought the current user can go to the show page of any public query and be able to at least run the SQL, but not be able to edit or delete the query. Is this correct?

**Comment by Javier-Jimenez-18** [comment]
- Modified Show, Edit, Update, and Destroy actions for the SavedQueriesController. Allowed a logged-in user to view the show page of another user's public query. 

Screen Shot 2022-12-16 at 1 51 36 AM

Screen Shot 2022-12-16 at 1 52 08 AM

Screen Shot 2022-12-16 at 1 53 31 AM

Screen Shot 2022-12-16 at 1 55 12 AM

Screen Shot 2022-12-16 at 1 55 25 AM

Screen Shot 2022-12-16 at 1 56 57 AM

Screen Shot 2022-12-16 at 1 57 43 AM

- Modified and updated the RSpec Controller test file.

Screen Shot 2022-12-16 at 1 50 22 AM

- Modified and updated the RSpec Request test file. Each test case PASSES when run individually. Will have to investigate further to see why the test file has FAILURES when all test cases are run together.

**Review by micronix — CHANGES_REQUESTED** [review]
The `show` action should have the logic to allow any user to see it if it is public or the user that created it is trying to see it.

The `edit`, `update` & `destroy` should only be accessible to the user that created it.

File: app/views/saved_queries/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    <div class="d-flex">
+      <div class="me-auto">
+        <button type="submit" class="btn btn-primary btn-sm">Execute SQL Query</button>
+      </div>
+      <div class="d-flex gap-3">
+        <div>
+          <%= link_to 'Edit', edit_saved_query_path(@saved_query), :class => "btn btn-primary btn-sm" %>
+        </div>
+        <div>
+          <%= link_to 'Delete', saved_query_path(@saved_query), :class => "btn btn-danger btn-sm", :method => :delete, data: { confirm: 'Are you sure?' } %>
```
Comment (micronix) [inline app/views/saved_queries/show.html.erb:36]: Only show the buttons if the logged in user is the one that created the saved query

**Review by micronix — CHANGES_REQUESTED** [review]

**Comment by Javier-Jimenez-18** [comment]
- Modified Show, Edit, and Destroy actions in the SavedQueriesController. 

- Modified Show view page to only show edit and delete buttons for the User that created the query.

Screen Shot 2022-12-16 at 10 12 29 AM

Screen Shot 2022-12-16 at 10 16 52 AM

Screen Shot 2022-12-16 at 10 13 01 AM

Screen Shot 2022-12-16 at 10 13 37 AM

Screen Shot 2022-12-16 at 9 15 38 AM

**Pull request merged**